solr/core/src/java/org/apache/solr/request/json/JsonQueryConverter.java

```diff
@@ -77,7 +84,10 @@ private void buildLocalParams(StringBuilder builder, Object val, boolean isQPars
 
       if (useSubBuilder) subBuilder = new StringBuilder();
 
-      subBuilder = subBuilder.append("{!").append(qtype).append(' ');;
+      if (tagName != null) {
+        subBuilder.append("{!tag=").append(tagName).append("}");
```

Comment:
I believe it should be quoted to avoid problems with spaces inside of tags. 
Also, would you mind to embed tag in the same ```{! }``` qparser, rather than wrap. I'm afraid of some edgecases.